### PR TITLE
[WIP] Process references

### PR DESCRIPTION
**Pull request by dalonsoa**
This adds the processing of the references depending on their type (plain, doi and bibkey).

I'm working in changing the current implementation to separate functions and adding tests, but I wanted to leave this here as a WIP to get some feedback from you all, specially from @ChasNelson1990 who was working on using bibtex.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
     author="Research Computing Service, Imperial College London",
     author_email="rcs-support@imperial.ac.uk",
     url="https://github.com/ImperialCollegeLondon/R2T2",
-    install_requires=["wrapt"],
+    install_requires=["wrapt", "bibtexparser"],
```
Comment (ChasNelson1990) [inline setup.py]: Have we not moved to poetry? Should this not go in pyproject.toml?

**Review by ChasNelson1990 — COMMENTED** [review]
Looks good to me. How were you thinking of splitting the functions? The existing implementation looks pretty efficient but it would be nice to be able to add a processor by just adding a function (like it was before I threw a spanner in the works!)

Otherwise, I only have an administrative comment.

File: setup.py (hunk 1 above)
Comment (jezcope) [reply setup.py]: It should, but #67 needs to be merged first.

**Comment by dalonsoa** [comment]
Massively stalled.

**Pull request closed without merging**